docs/concepts/controllers/statefulsets.md

```diff
@@ -0,0 +1,171 @@
+be certain that a [Deployment](/docs/user-guide/deployments/) or a
+[ReplicaSet](/docs/user-guide/replicasets/) is not better suited to your needs.
+
+### Limitations
+* StatefulSet is a beta resource, not available in any Kubernetes release prior to 1.5.
+* As with all alpha/beta resources, it can be disabled through the `--runtime-config` option passed to the apiserver.
+* The storage for a given Pod must either be provisioned by a [PersistentVolume Provisioner](http://releases.k8s.io/{{page.githubbranch}}/examples/experimental/persistent-volume-provisioning/README.md) based on the requested `storage class`, or pre-provisioned by an admin.
+* Deleting and/or scaling a StatefulSet down will *not* delete the volumes associated with the StatefulSet. This is done to ensure safety first, your data is more valuable than an auto purge of all related StatefulSet resources. 
+* StatefulSets currently require a [Headless Service](/docs/user-guide/services/#headless-services) to be responsible for the network identity of the Pods. The user is responsible for this Service.
+* Updating an existing StatefulSet is currently a manual process.
+
+### Components
+The example below demonstrates the components of a StatefulSet. 
+
+* A Headless Service, named nginx, is used to control the network domain. 
+* The StatefulSet, named web, has a Spec that indicates that 3 replicas of the nginx container will be launched in unique Pods.
+* The volumeClaimTemplates, will provide stable storage using [PersistentVolumes](/docs/user-guide/volumes/) provisioned by a 
+ PersistentVolume Provisioner.
+
+```yaml
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: nginx
+  labels:
+    app: nginx
+spec:
+  ports:
+  - port: 80
+    name: web
+  clusterIP: None
+  selector:
+    app: nginx
+---
+apiVersion: apps/v1beta1
+kind: StatefulSet
+metadata:
+  name: web
+spec:
+  serviceName: "nginx"
+  replicas: 3
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      terminationGracePeriodSeconds: 10
+      containers:
+      - name: nginx
+        image: gcr.io/google_containers/nginx-slim:0.8
+        ports:
+        - containerPort: 80
+          name: web
+        volumeMounts:
+        - name: www
+          mountPath: /usr/share/nginx/html
+  volumeClaimTemplates:
+  - metadata:
+      name: www
+    spec:
+      accessModes: [ "ReadWriteOnce" ]
+      resources:
+        requests:
+          storage: 1Gi
+```
+
+### Pod Identity
+StatefulSet Pods have a unique identity that is comprised of an ordinal, a 
+stable network identity, and stable storage. The identity sticks to the Pod, 
+regardless of which node it's (re) scheduled on.
+
+__Ordinal Index__
+
+For a StatefulSet with N replicas, each Pod in the StatefulSet will be 
+assigned an integer ordinal, in the range [0,N), that is unique over the Set. 
+
+__Stable Network ID__
+
+The hostname of a Pod in a StatefulSet is derived from the name of the 
```

Comment:
"Each Pod in a StatefulSet derives its hostname from the name of the StatefulSet and the ordinal of the Pod."